examples/classification/classification/evaluator_multiclass.py

```diff
@@ -0,0 +1,261 @@
+from classification.workflow import ThresholdConfiguration
+
+from kolena._experimental.classification.utils import compute_confusion_matrix
+from kolena._experimental.classification.utils import compute_roc_curves
+from kolena._experimental.classification.utils import create_histogram
+from kolena._experimental.classification.utils import get_histogram_range
+from kolena._utils import log
+from kolena.workflow import EvaluationResults
+from kolena.workflow import Plot
+from kolena.workflow import TestCases
+from kolena.workflow.metrics import accuracy as compute_accuracy
+from kolena.workflow.metrics import f1_score as compute_f1_score
+from kolena.workflow.metrics import precision as compute_precision
+from kolena.workflow.metrics import recall as compute_recall
+from kolena.workflow.plot import Histogram
+
+
+def compute_test_sample_metric(
+    ground_truth: GroundTruth,
+    inference: Inference,
+    threshold_configuration: ThresholdConfiguration,
+) -> TestSampleMetrics:
+    empty_metrics = TestSampleMetrics(
+        classification=None,
+        margin=None,
+        is_correct=False,
+    )
+
+    if len(inference.inferences) == 0:
+        return empty_metrics
+
+    sorted_infs = sorted(inference.inferences, key=lambda x: x.score, reverse=True)
+    predicted_match = sorted_infs[0]
+    predicted_label, predicted_score = predicted_match.label, predicted_match.score
+
+    if threshold_configuration.threshold is not None and predicted_score < threshold_configuration.threshold:
+        return empty_metrics
+
+    return TestSampleMetrics(
+        classification=predicted_match,
+        margin=predicted_score - sorted_infs[1].score if len(sorted_infs) >= 2 else None,
+        is_correct=predicted_label == ground_truth.classification.label,
+    )
+
+
+def compute_test_case_metrics(
+    ground_truths: List[GroundTruth],
+    metrics_test_samples: List[TestSampleMetrics],
+    labels: List[str],
+) -> TestCaseMetrics:
+    classification_pairs = [
+        (gt.classification.label, tsm.classification.label if tsm.classification else None)
+        for gt, tsm in zip(ground_truths, metrics_test_samples)
+    ]
+    n_images = len(classification_pairs)
+    class_level_metrics: List[ClassMetricsPerTestCase] = []
+    for label in sorted(labels):
+        n_tp = len([True for gt, inf in classification_pairs if gt == label and inf == label])
+        n_fn = len([True for gt, inf in classification_pairs if gt == label and inf != label])
+        n_fp = len([True for gt, inf in classification_pairs if gt != label and inf == label])
+        n_tn = len([True for gt, inf in classification_pairs if gt != label and inf != label])
+
+        class_level_metrics.append(
+            ClassMetricsPerTestCase(
+                label=label,
+                TP=n_tp,
+                FP=n_fp,
+                FN=n_fn,
+                TN=n_tn,
+                Accuracy=compute_accuracy(n_tp, n_fp, n_fn, n_tn),
```

Comment:
Accuracy might be a misleading metric -- if I have 10 classes and I have a wrong inference, that still means that it's an FP for one label, a FN for another, and for the other 8, it counts as a `TN`. That means terrible models would still have an accuracy of > 0.8. Is this standard in the approach?

Reply:
Correct. It can be a misleading metric in some context but it's hard to drop it as it is still one of the standard metrics in Classification workflow.  For example, sklearn has [`classification_report` tool](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.classification_report.html) that summarizes the classification metrics and it includes `accuracy` as well.